src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpClientJsonExtensions.Get.AsyncEnumerable.cs

```diff
@@ -135,12 +135,29 @@ public static partial class HttpClientJsonExtensions
             JsonSerializerOptions? options,
             CancellationToken cancellationToken)
         {
-            options ??= JsonSerializerOptions.Default;
-            options.MakeReadOnly();
+            if (client is null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
 
-            var jsonTypeInfo = (JsonTypeInfo<TValue>)options.GetTypeInfo(typeof(TValue));
+            return Core(client, requestUri, options, cancellationToken);
 
-            return FromJsonStreamAsyncCore(client, requestUri, jsonTypeInfo, cancellationToken);
+            static async IAsyncEnumerable<TValue?> Core(
```

Comment:
Likewise, this change would need to be reverted.

Reply:
Reverted